### PR TITLE
Adding type annotations

### PR DESCRIPTION
**Pull request by fivesixzero**
This PR includes type annotaiton additions with a few related changes to address #30.

These are the two related changes made, beyond the addition of type annotations:

1. Moved pre-defined "rotated keymap" tuples from in-line constants into module level constants
    * The zero-rotation default is used in two places so it seemed sensible to just move all of these constants up into the module.
2. Added a `pylint` disable for `too-many-lines`
    * These changes pushe the module's line count to 1,005, just over the default max of `1000`

### Testing

After the changes `pre-commit` completes without issue. `mypy --disallow-untyped-defs` returns some errors (noted below) but with none related to missing type or return type annotations.

To be thorough I tested the modified library using each of the examples on a Macropad running the latest CircuitPython release (`7.1.1`). All of the examples ran as expected without issues or errors.

I didn't do additional testing to check library size or memory footprint deltas but those should be minimal and the RP2040 offers quite a bit of margin there, so we should be fine.

### Potential Loose Ends

There are a still a few remaining `mypy` typing errors related to some class vars being instantiated with `None` and being assigned later.

```
adafruit_macropad.py:818: error: Incompatible types in assignment (expression has type "array[int]", variable has type "None")
adafruit_macropad.py:870: error: "None" has no attribute "sample_rate"
adafruit_macropad.py:870: error: Argument 1 to "len" has incompatible type "None"; expected "Sized"
adafruit_macropad.py:871: error: "None" has no attribute "playing"
adafruit_macropad.py:872: error: "None" has no attribute "play"
```

If addressing those is important to this effort I'll follow up with a second commit. I'm not sure what the best way is to address those aside from at least a bit of minor refactoring so any advice, guidance, or examples for a quick fix would be helpful if this is required for fully addressing the type annotations issue. 😄

### CODE REVIEW

**Comment by fivesixzero** [comment]
Pushed a second commit to fix an error with Sphinx's annotation parsing.

The error was related to Python 3's annotation parsing, described in this `mypy` doc:

https://mypy.readthedocs.io/en/stable/runtime_troubles.html

TL;DR is that a class declared after its annotation isn't 'valid'.

This was addressed in Python 3.11's annotation handler, so adding `from __future__ import annotations` fixes this when using a full Python interpreter. But this import needs to happen at the top of the file and can't be done within our handy typing `try`/`except` block, which looks like it could cause issues with CircuitPython (although admittedly I didn't test this).

Pre-3.11 the most simple fix looks like moving things around to declare the class prior to any annotation(s) using said class.

In the end I chose to simply move the `_PixelMapLite` class above the `MacroPad` class in the module which addressed the Sphinx issue at the cost of breaking a cardinal readability rule.

If there are better ways of fixing this I'll be happy to submit another commit with those changes. 😄

**Review by PaintYourDragon — APPROVED** [review]

**Comment by fivesixzero** [comment]
I hope adding a fix for #35 (a76f87e) doesn't complicate matters with this type annotation PR but it seemed simple enough to drop in the 3-line change to avoid having to go through a separate review cycle.

If that needs to be split out into a separate PR just let me know and I'll take care of it. 😄
